Source/DiabloUI/diabloui.cpp

```diff
@@ -463,6 +463,7 @@ void UiHandleEvents(SDL_Event *event)
 		return;
 	}
 
+#if HAS_KBCTRL == 0

```

Comment:
This file needs to include `controls/devices/kbcontroller.h`, which sets `HAS_KBCTRL` to `0` if it isn't defined

Reply:
ah I tested under #ifndef at first, then decided to use #if 0